libs/generic-view/store/src/lib/modals/reducer.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { createReducer } from "@reduxjs/toolkit"
+import {
+  closeAllModals,
+  closeDomainModals,
+  closeModal,
+  openModal,
+  replaceModal,
+} from "./actions"
+
+export interface Modal {
+  key: string
+  domain?: string
+  replaced?: boolean
```

Comment:
This prop helps in situation when modal replaces another one, and then the new one is closed, so the previous (with replaced: true) can be shown again.